src/login/NativeWebauthnView.ts

```diff
@@ -0,0 +1,52 @@
+import type {Children, Vnode} from "mithril"
+import m from "mithril"
+import type {CurrentView} from "../gui/base/Header.js"
+import {DialogHeaderBar, DialogHeaderBarAttrs} from "../gui/base/DialogHeaderBar.js"
+import type {WebauthnNativeBridge} from "../native/main/WebauthnNativeBridge"
+import {IWebauthn} from "../misc/2fa/webauthn/IWebauthn.js"
+import {SecondFactorImage} from "../gui/base/icons/Icons.js"
+import {progressIcon} from "../gui/base/Icon.js"
+import {lang} from "../misc/LanguageViewModel.js"
+import {ButtonType} from "../gui/base/ButtonN.js"
+
+/** This is a special view which is not used by the web client directly but is loaded remotely by desktop client in a dialog. See DesktopWebauthn. */
+export class NativeWebauthnView implements CurrentView {
+	constructor(
+		private readonly webauthn: IWebauthn,
+		private readonly nativeTransport: WebauthnNativeBridge
+	) {
+		this.view = this.view.bind(this)
```

Comment:
Maybe a comment why this is still necessary

Reply:
The same reason as we do it in all old-style components?

Reply:
that's not very descriptive though, I was thinking it should say more _why_ we need to make it like that in the first place, i.e. because it's a top-level view and that's how they work currently. Maybe it's obvious though